### PR TITLE
Feat/new flavor scheme

### PR DESCRIPTION
**Pull request by garloff**
This is a first version how the new flavor naming v2 could look like to address issue #190, reflecting the considerations from #192.
A document that describes how the transition can be done smoothly still needs to be written.
The check tools which has been updated along with the standard can be run in a mode to accept both old v1 and new v2 flavor names, which hints how a transition might be implemented.
To check for v1 compliance, the tool has an option `-1` now, so we don't need to carry an old version of the test tool.

### CODE REVIEW

File: Tests/iaas/flavor-naming/flavor-name-check.py [hunk 1]
```diff
@@ -27,11 +27,14 @@
 import sys
 import os
 import re
+# import getopt
```
Comment (fkr) [inline Tests/iaas/flavor-naming/flavor-name-check.py:30]: unused, commented out import

File: Tests/iaas/flavor-naming/flavor-name-check.py [hunk 2]
```diff
@@ -217,9 +251,10 @@ def out(self):
                     i += 3
                     lst.append(att)
                 par += 1
-            elif self.outstr[i+1] == ":":
+            elif self.outstr[i+1] == ":":   # change?
                 if att:
-                    ostr += ":%"
+                    # ostr += ":%"
```
Comment (fkr) [inline Tests/iaas/flavor-naming/flavor-name-check.py:256]: imho remove, if left in commented out, imho leave comment why it is left there, so it clear why it was left there.

**Review by fkr — APPROVED** [review]
LGTM! left two very minor comments

File: Tests/iaas/flavor-naming/flavor-name-check.py (hunk 2 above)
Comment (garloff) [reply Tests/iaas/flavor-naming/flavor-name-check.py]: Fair comment, this has changed from v1 to v2. But git history is good enough if we ever want to do archeology.

File: Tests/iaas/flavor-naming/flavor-name-check.py (hunk 1 above)
Comment (garloff) [reply Tests/iaas/flavor-naming/flavor-name-check.py]: Removed.
Will readd in a subsequent patch, as the hand-written option parsing needs to go.

**Review by tibeer — APPROVED** [review]
LGTM! Just one hint: I have the feeling that the `flavor-name-check.py` script might become too complex to be easily maintained. That being said, I do not have any good idea on how to solve this.

**Review by wwentland — APPROVED** [review]
LGTM, thank you for addressing my concerns. Much appreciated!

**Review by fkr — APPROVED** [review]
LGTM

**Review by itrich — APPROVED** [review]
LGTM

**Pull request merged**